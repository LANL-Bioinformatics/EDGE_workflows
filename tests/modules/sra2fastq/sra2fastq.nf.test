nextflow_workflow {

    name "Test Workflow SRA2FASTQ"
    script "modules/sra2fastq/sra2fastq.nf"
    workflow "SRA2FASTQ"

    test("Basic workflow") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success },
                {assert snapshot(workflow.out).match()}
            )
        }

    }
    
    test("Ensure ILLUMINA platform") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    platformRestrict = "Illumina"
                    accessions = ["SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()}
            )
        }

    }

    test("Ensure not PACBIO_SMRT platform") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    platformRestrict = "PACBIO_SMRT"
                    accessions = ["SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert workflow.out.paired.size() == 0},
                {assert workflow.out.unpaired.size() == 0},
                {assert snapshot(workflow.out).match()}
            )
        }

    }
    test("Ensure not OXFORD_NANOPORE platform") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    platformRestrict = "OXFORD_NANOPORE"
                    accessions = ["SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert workflow.out.paired.size() == 0},
                {assert workflow.out.unpaired.size() == 0},
                {assert snapshot(workflow.out).match()}
            )
        }

    }
    test("Only Download Unique Accessions") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562","SRR29462562","SRR29462562","SRR29462562","SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.trace.succeeded().size() == 1},
                {assert workflow.out.unpaired.size() == 1}
            )
        }

    }
    test("Clean Output") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562"]
                    clean = true
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()}
            )
        }

    }
    test("Multiple accessions") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR11241255", "SRX20564104", "SRR29462562", "SRX24960352"]
                    clean = true
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.out.unpaired.size() == 2},
                {assert workflow.out.paired.size() == 2}
            )
        }

    }
    test("Filtered Accessions") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR11241255", "SRX20564104", "SRR29462562", "SRX24960352", "SRX26473729"]
                    platformRestrict = "PACBIO_SMRT"
                    clean = true
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.out.unpaired.size() == 1}
            )
        }

    }
}
