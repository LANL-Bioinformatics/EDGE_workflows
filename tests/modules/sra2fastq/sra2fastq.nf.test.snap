{
    "Multiple accessions": {
        "content": [
            {
                "0": [
                    [
                        "SRX20564104_1.fastq.gz:md5,4e4c4f89c097d782364f55a4828ba4c3",
                        "SRX20564104_2.fastq.gz:md5,0a56586f63e03fc3c9d2cb26213b80f0"
                    ],
                    [
                        "SRX24960352_1.fastq.gz:md5,e869779afbd424971f7e41aa342836cc",
                        "SRX24960352_2.fastq.gz:md5,cb316c949fcb308017bbfe8ff5863752"
                    ]
                ],
                "1": [
                    "SRR11241255.fastq.gz:md5,2df0fa97a124de8b8d4d939f024bd594",
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "paired": [
                    [
                        "SRX20564104_1.fastq.gz:md5,4e4c4f89c097d782364f55a4828ba4c3",
                        "SRX20564104_2.fastq.gz:md5,0a56586f63e03fc3c9d2cb26213b80f0"
                    ],
                    [
                        "SRX24960352_1.fastq.gz:md5,e869779afbd424971f7e41aa342836cc",
                        "SRX24960352_2.fastq.gz:md5,cb316c949fcb308017bbfe8ff5863752"
                    ]
                ],
                "unpaired": [
                    "SRR11241255.fastq.gz:md5,2df0fa97a124de8b8d4d939f024bd594",
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:05:34.897015"
    },
    "Filtered Accessions": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "SRX26473729.fastq.gz:md5,64e17afb49765db68336dbc9669f8a0d"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "SRX26473729.fastq.gz:md5,64e17afb49765db68336dbc9669f8a0d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:56:11.003119"
    },
    "Ensure not OXFORD_NANOPORE platform": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T17:03:58.552875"
    },
    "Only Download Unique Accessions": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:03:32.397274"
    },
    "Ensure ILLUMINA platform": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:01:45.248109"
    },
    "Ensure not PACBIO_SMRT platform": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T17:03:47.363828"
    },
    "Clean Output": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:04:16.360353"
    },
    "Basic workflow": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-23T16:01:03.780856"
    }
}